code/ATMOSPHERICS/components/unary_devices/vent_pump.dm

```diff
@@ -381,6 +381,8 @@
 	if(!welded)
 		if(open)
 			for(var/obj/item/W in src)
+				if(istype(W, /obj/item/pipe))
```

Comment:
So the only thing that can be contained in vents like this are papers. It would be better if you created a holder var for papers and did a minor refactor so that papers moved into these vents are referenced there, then you can check for that paper holder var, explicitly, rather than relying on this blind check (which is awful).

Reply:
I mean yes, but then you would need extra handling for when the vent is unwrenched or destroyed. And it would be exempt from procs that are acting on the contents of the vent (though I can't think of any off the top of my head).
Having it in "contents" would be neater.

Do you think it might be better to just not have the vent contain its "/obj/item/pipe" as an item?